client/js/table-view.js

```diff
@@ -78,15 +80,43 @@ class TableView {
 		this.sheetBodyEl.appendChild(fragment);
 	}
 
+	renderTableFooter() {
+		const fragment = document.createDocumentFragment();
+		let total = {};
```

Comment:
Cool memoization. However, "total" doesn't tell me (another dev) that you are using this object to store the sums of columns you've already added. Use a descriptive name that communicates what this object holds.